dnf-behave-tests/dnf/plugins-core/config-manager-addrepo.feature

```diff
@@ -0,0 +1,437 @@
+        """
+   When I execute dnf with args "config-manager addrepo --from-repofile={context.dnf.tempdir}/tmp/test.repo --create-missing-dir"
+   Then the exit code is 0
+    And file "/etc/yum.repos.d/test.repo" contents is
+        """
+        [test]
+        name=repository file
+        enabled=0
+        baseurl=http://something.com/os/
+        """
+
+
+Scenario: test "addrepo" from repofile, a file was found instead of destination directory
+ Given I delete directory "/etc/yum.repos.d/"
+    And I create file "/etc/yum.repos.d" with
+    """
+    """
+    And I create file "/{context.dnf.tempdir}/tmp/test.repo" with
+        """
+        [test]
+        name=repository file
+        enabled=0
+        baseurl=http://something.com/os/
+        """
+   When I execute dnf with args "config-manager addrepo --from-repofile={context.dnf.tempdir}/tmp/test.repo"
+   Then the exit code is 1
+    And stderr contains "The path ".*/etc/yum.repos.d" exists, but it is not a directory or a symlink to a directory"
+
+
+Scenario: test "addrepo" from repofile, a symlink to non-existent object was found instead of destination directory
+ Given I delete directory "/etc/yum.repos.d/"
+    And I create symlink "/etc/yum.repos.d" to file "/non-exist"
+    And I create file "/{context.dnf.tempdir}/tmp/test.repo" with
+        """
+        [test]
+        name=repository file
+        enabled=0
+        baseurl=http://something.com/os/
+        """
+   When I execute dnf with args "config-manager addrepo --from-repofile={context.dnf.tempdir}/tmp/test.repo"
+   Then the exit code is 1
+    And stderr contains "The path ".*/etc/yum.repos.d" exists, but it is a symlink to a non-existent object"
+
+
+Scenario: test "addrepo" from repofile, destination repo file already exists
+ Given I create file "/{context.dnf.tempdir}/tmp/repo1.repo" with
+        """
+        [repo1]
+        name=repository file
+        enabled=0
+        baseurl=http://something.com/os/
+        """
+   When I execute dnf with args "config-manager addrepo --from-repofile={context.dnf.tempdir}/tmp/repo1.repo"
+   Then the exit code is 1
+    And stderr contains "File \".*/etc/yum.repos.d/repo1.repo\" already exists and configures repositories with IDs \"repo1\". Add "--overwrite" to overwrite"
+
+
+Scenario: test "addrepo" from repofile, repository id already exists
+ Given I create file "/{context.dnf.tempdir}/tmp/test.repo" with
+        """
+        [repo1]
+        name=repository file
+        enabled=0
+        baseurl=http://something.com/os/
+        """
+   When I execute dnf with args "config-manager addrepo --from-repofile={context.dnf.tempdir}/tmp/test.repo"
+   Then the exit code is 1
+    And stderr contains "A repository with id \"repo1\" already configured in file: .*/etc/yum.repos.d/repo1.repo"
+
+
+Scenario: test "addrepo" from repofile, overwrite existing repo file (--overwrite), the repo id already exists but in the overwritten file so no problem
```

Comment:
May I ask you to add a comment or a test what is the starting point. In Background step it creates repo1.repo with repoid `repo1`.

Reply:
Added